### ISSUE TITLE
RuntimeWarning: Mean of empty slice.

### ISSUE THREAD

**Issue by zhjmcjk** [automl/auto-sklearn#600]
**I try to run the following example:**
`import autosklearn.classification`
`import sklearn.model_selection`
`import sklearn.datasets`
`import sklearn.metrics`
`X, y = sklearn.datasets.load_digits(return_X_y=True)`
`X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(X, y, random_state=1)`
`automl = autosklearn.classification.AutoSklearnClassifier()`
`automl.fit(X_train, y_train)`
`y_hat = automl.predict(X_test)`
`print("Accuracy score", sklearn.metrics.accuracy_score(y_test, y_hat))`

**The program keeps printing the warning as follows:**
![image](https://user-images.githubusercontent.com/31545030/50153415-4f316a80-0301-11e9-8bde-8601a1b03815.png)
![image](https://user-images.githubusercontent.com/31545030/50153425-535d8800-0301-11e9-9018-48d864f98d17.png)
![image](https://user-images.githubusercontent.com/31545030/50153430-56587880-0301-11e9-8ac1-21cdda1c7bf1.png)

**My system details:**
ubuntu 16.04 LTS
python 3.6
All dependency packages meet this requirement: https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt

**Comment by NothingLZ** [comment]
me too

**Comment by mfeurer** [comment]
These are caused by a bug computing the training error, which is not used for anything (and is only calculated to be analyzed afterwards). Therefore, this does not harm the performance of Auto-sklearn.

**Comment by NothingLZ** [comment]
@mfeurer How to solve it?

**Comment by mfeurer** [comment]
I already assigned @ahn1340 to this so it is best to wait for this to be closed.

**Comment by ajoeajoe** [comment]
I wonder if this Warning solved？

**Comment by mfeurer** [comment]
@ahn1340 are there any news on this?

**Comment by iiSeymour** [comment]
Until #679 you can reduce the noise with `filterwarnings`.

```python
import warnings
warnings.filterwarnings("ignore", "Mean of empty slice")
```

**Comment by ahn1340** [comment]
Hi, sorry for the awfully late reply. I made a PR which rewrites the computation of training loss and would make the warning go away. I will ask @mfeurer to have a look at the PR and merge it to the dev branch.

**Comment by mfeurer** [comment]
I finally merged this. It'll be fixed in the next version (probably being released next week).
